controllers/hostedcontrolplane/hostedcontrolplane.go

```diff
@@ -323,3 +358,144 @@ func (r *HostedControlPlaneReconciler) SetupWithManager(mgr ctrl.Manager) error
+}
+
+func (r *HostedControlPlaneReconciler) UpdateHostedControlPlaneLabels(ctx context.Context, hostedcontrolplane *hypershiftv1beta1.HostedControlPlane, key, value string) error {
+	labels := hostedcontrolplane.GetLabels()
+	labels[key] = value
+	hostedcontrolplane.SetLabels(labels)
+
+	err := r.Client.Update(ctx, hostedcontrolplane)
+	if err != nil {
+		return fmt.Errorf("error updating hostedcontrolplane monitor: %v", err)
+	}
+	return nil
+}
+
+func GetAPIServerHostname(hostedcontrolplane *hypershiftv1beta1.HostedControlPlane) (string, error) {
+	for _, service := range hostedcontrolplane.Spec.Services {
+		if service.Service == "APIServer" {
+			return service.ServicePublishingStrategy.Route.Hostname, nil
+		}
+	}
+	return "", fmt.Errorf("APIServer service not found in the hostedcontrolplane")
+}
+
+func (r *HostedControlPlaneReconciler) deployDynatraceHTTPMonitorResources(DynatraceAPIClient *dynatrace.DynatraceAPIClient, ctx context.Context, log logr.Logger, hostedcontrolplane *hypershiftv1beta1.HostedControlPlane) error {
+	//if http monitor does not exist, and hcp is not marked for deletion, and hcp is ready, then create http monitor
+	//get apiserver
+	APIServerHostname, err := GetAPIServerHostname(hostedcontrolplane)
+	if err != nil {
+		return fmt.Errorf("failed to get APIServer hostname %v", err)
+	}
+	monitorName := strings.Replace(APIServerHostname, "api.", "", 1)
+	// APIServerHostname := hostedcontrolplane.Spec.Services[1].ServicePublishingStrategy.Route.Hostname
+	monitorLocation := hostedcontrolplane.Spec.Platform.AWS.EndpointAccess
+
+	//in hcp, spec.services.service["APIServer"].servicePublishingStrategy.route.hostname is api.test-rs1.dgcj.i3.devshift.org
+	// apiUrl := "https://api.hb-testing.j1b6.i3.devshift.org/livez"
+
+	apiUrl := fmt.Sprintf("https://%s/livez", APIServerHostname)
+
+	dynatraceHttpMonitorId, err := getDynatraceHTTPMonitorID(hostedcontrolplane)
+	if err != nil {
+		log.Info(fmt.Sprintf("error calling getDynatraceHTTPMonitorID %v", err))
+	}
+
+	if dynatraceHttpMonitorId != "" {
+		log.Info("HTTP monitor Found. Skipping creating a monitor")
+		return nil
+	}
+	// determine location and create monitor
+	//public
+	if monitorLocation == "PublicAndPrivate" {
+
+		clusterID := hostedcontrolplane.Spec.ClusterID
+		clusterRegion := hostedcontrolplane.Spec.Platform.AWS.Region
+
+		dynatraceEquivalentClusterRegionId, err := DynatraceAPIClient.GetDynatraceEquivalentClusterRegionId(clusterRegion)
+		if err != nil {
+			return fmt.Errorf("error getting DynatraceEquivalentClusterRegionId: %v", err)
+		}
+
+		monitorID, err := DynatraceAPIClient.CreateDynatraceHTTPMonitor(monitorName, apiUrl, clusterID, dynatraceEquivalentClusterRegionId)
+		if err != nil {
+			return fmt.Errorf("error creating HTTP monitor: %v", err)
+		}
+
+		err = r.UpdateHostedControlPlaneLabels(ctx, hostedcontrolplane, httpMonitorLabel, monitorID)
+		if err != nil {
+			return fmt.Errorf("failed to update hostedcontrolplane monitor labels %v", err)
```

Comment:
If we run into this error, I believe we would start creating synthetic monitors in Dynatrace over and over again right?

Reply:
Good catch! I think we can call `dynatraceApiClient.DeleteDynatraceHttpMonitor(monitorId)` within `if err!=nil` here which should take care of this issue right?

Reply:
Working as expected - 
```go
if err != nil {
			deleteErr := dynatraceApiClient.DeleteDynatraceHttpMonitor(monitorId)
			if deleteErr != nil {
				log.Error(deleteErr, "error deleting HTTP monitor")
			}
			return fmt.Errorf("failed to update hostedcontrolplane monitor labels %v", err)
		}
		```

Reply:
Now that we have https://github.com/openshift/route-monitor-operator/pull/311/files#diff-4901142434d44dc34c348a0ea0603548e55e4dd41dee8217bc9f8216101d68d4R436-R446, we don't need to delete the monitor anymore right? We can return an error and the next reconcile will try again.

```go
if err != nil {
    return fmt.Errorf("failed to update hostedcontrolplane monitor labels %v", err)
}
```